pkg/controller/appmetricsapp/appmetricsapp_controller.go

```diff
@@ -0,0 +1,204 @@
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileAppMetricsApp{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("appmetricsapp-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource AppMetricsApp
+	err = c.Watch(&source.Kind{Type: &metricsv1alpha1.AppMetricsApp{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner AppMetricsApp
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &metricsv1alpha1.AppMetricsApp{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileAppMetricsApp implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileAppMetricsApp{}
+
+// ReconcileAppMetricsApp reconciles a AppMetricsApp object
+type ReconcileAppMetricsApp struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a AppMetricsApp object and makes changes based on the state read
+// and what is in the AppMetricsApp.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileAppMetricsApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling AppMetricsApp")
+
+	// Fetch the AppMetricsApp instance
+	instance := &metricsv1alpha1.AppMetricsApp{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	is, err := IsValidAppNamespace(instance.ObjectMeta.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !is {
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	routeList := &routev1.RouteList{}
+	listOptions := &client.ListOptions{}
+	listOptions.InNamespace("app-metrics")
```

Comment:
Rather than hard-coding `"app-metrics"` here, I'd use `k8sutil.GetOperatorNamespace()` (not sure what that function does for `operator-sdk up local` though).

Reply:
Done